test-framework/src/main/java/com/datastax/oss/quarkus/test/CassandraTestResource.java

```diff
@@ -13,23 +13,43 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.datastax.oss.quarkus.tests;
+package com.datastax.oss.quarkus.test;
 
 import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+import java.net.URL;
 import java.util.Collections;
 import java.util.Map;
 import org.jboss.logging.Logger;
 import org.testcontainers.containers.CassandraContainer;
 import org.testcontainers.containers.wait.CassandraQueryWaitStrategy;
 
+/**
+ * The integration test that is using this resources should have defined two settings in the
+ * .properties file:
+ *
+ * <pre>
+ * quarkus.cassandra.contact-points=127.0.0.1:${quarkus.cassandra.docker_port}
+ * quarkus.cassandra.load-balancing-policy.local-datacenter=datacenter1
+ * </pre>
+ *
+ * Please note the ${quarkus.cassandra.docker_port} - it will be automatically injected by the
+ * {@code CassandraTestResource}.
+ *
+ * <p>If you want to execute the CQL init logic (i.e. CREATE KEYSPACE or CREATE TABLE) please create
+ * an init_script.cql file and put it in the resources folder.
+ */
 public class CassandraTestResource implements QuarkusTestResourceLifecycleManager {
   private static final Logger LOGGER = Logger.getLogger(CassandraTestResource.class);
   private static CassandraContainer<?> cassandraContainer;
 
   @Override
   public Map<String, String> start() {
     cassandraContainer = new CassandraContainer<>();
-    cassandraContainer.withInitScript("init_script.cql");
+    // set init script only if it's provided by the caller
+    URL resource = Thread.currentThread().getContextClassLoader().getResource("init_script.cql");
```

Comment:
Are you sure? Why not simply `getClass().getResource(...)`?

Reply:
yes, this is how the TestContainers is loading the initSript (I was surprised by that)